tests/common/plugins/sanity_check/checks.py

```diff
@@ -100,41 +100,58 @@ def check_interfaces(dut):
 def check_bgp_status(dut):
 
     def _check_bgp_status_helper():
-        neighbors.clear()
-        neighbors.update(dut.get_bgp_neighbors())
-        if neighbors:
-            down_neighbors = [k for k, v in neighbors.items()
-                              if v['state'] != 'established']
-            if down_neighbors:
-                check_result['down_neighbors'] = down_neighbors
-                check_result['failed'] = True
+        asic_check_results = []
+        bgp_facts = dut.bgp_facts(asic_index='all')
+        for asic_index, a_asic_facts in enumerate(bgp_facts):
+            a_asic_result = False
+            a_asic_neighbors = a_asic_facts['ansible_facts']['bgp_neighbors']
+            if a_asic_neighbors:
+                down_neighbors = [k for k, v in a_asic_neighbors.items()
+                                  if v['state'] != 'established']
+                if down_neighbors:
+                    if dut.facts['num_asic'] == 1:
+                        check_result['bgp'] = {'down_neighbors' : down_neighbors }
+                    else:
+                        check_result['bgp' + str(asic_index)] = {'down_neighbors' : down_neighbors }
+                    a_asic_result = True
+                else:
+                    a_asic_result = False
+                    if dut.facts['num_asic'] == 1:
+                        if 'bgp' in check_result:
```

Comment:
If we have 2 devices, dut1 and dut2.  dut1 has some down_neighbors, dut2 has no down_neighbors. 
the down_neighbors will be removed from check_results here. Is this correct ?

Reply:
No. The final results is a dictionary with key being each DUT's hostname, and value a list of all sanity checks on that DUT. So, for your scenario, check_results would be something like:

```
{
    "node1": [
        ....
        {
            "check_item": "bgp", 
            "failed": true, 
            "bgp": {
                "down_neighbors": [
                    "2064:101::1", 
                    "100.0.0.1", 
                    "101.0.0.1", 
                    "2064:100::1"
                ]
            }
        }, 
        .....
   "node2": [
        .....
        {
            "check_item": "bgp", 
            "failed": false, 
        }, 
        .....
     ]
  }, 
  

```